### DISCUSSION TITLE
Mutating image data does have any effect

### DISCUSSION THREAD

**Discussion by danielwolbach** [bevyengine/bevy#12122] [Q&A] [answered]
Hi everyone! I am new to Bevy and I am am trying to achieve this: I have a constant stream of `[u8]` pixel data and I want to render/update it using Bevy. I create and store a `Handle<Image>` and mutate the data directly like so:
```rust
fn update_texture(
    time: Res<Time>,
    mut timer: ResMut<TextureUpdateTimer>,
    query: Query<&View>,
    mut images: ResMut<Assets<Image>>,
) {
    if timer.0.tick(time.delta()).just_finished() {
        for view in query.iter() {
            if let Some(image) = images.get_mut(&game_view.image_handle) {
                image.data = vec![255; 160 * 144 * 4];
            }
        }
    }
```
Unfortunately, the new image does not show. I created the image using `RenderAssetUsages::RENDER_WORLD | RenderAssetUsages::MAIN_WORLD`. Please let me know if you have any ideas or advice :) Thanks!

**Comment by mosure** [comment]
I found the material which uses the image also needs to be updated (with the same `Handle<Image>`), e.g.

```rust
pub struct CameraMaterial {
    #[texture(0)]
    #[sampler(1)]
    source_image: Handle<Image>,
}


...


let image = images.get_mut(image_handle.clone()).unwrap();
image.data = decoded.to_vec();

let material = camera_materials.get_mut(material_handle).unwrap();
material.source_image = image_handle;
```

**Reply by newDINO** [reply]
Actually no need to replace the image handle. Just get the mutable reference of the material also works.

**Comment by danielwolbach** [comment, marked as answer]
I was able to fix it by using a `SpriteBundle` instead of a  `PbrBundle`. The `SpriteBundle` is sufficient for me but I don't really know why it didn't work with the `PbrBundle`.
